CMSgov/mint-app
# Easi 1941/participants and providers

# EASI-1941

## Changes and Description

- Implemented Participants and Providers Task list Section
- Implemented generic status calculation for this new section

--> GQL
--> Dev Data seeding
--> Unit Tests

## How to test this change
1. Confirm Unit Tests
2. Compare data fields against FIGMA, looking for any fields that might have been missed
3. Try to update the task list section ( steps below)

run `scripts/dev db:seed`
--> You can now run the code below to update the seeded Participants and Providers data
```gql
mutation updatePlanParticipantsAndProviders #($input: PlanMilestonesInput!) 
{
    updatePlanParticipantsAndProviders (id: "da79bad2-b4ce-4975-80bf-bd6e71ae300f",changes:     
    {


    #Page 1
    # participants:
    # medicareProviderType:
    statesEngagement: "Very good engagement"
    # participantsOther:
    # participantsNote:
    participantsCurrentlyInModels: null
    # participantsCurrentlyInModelsNote:
    # modelApplicationLevel:

    #Page 2
    # expectedNumberOfParticipants:
    # estimateConfidence:
    # confidenceNote:
    # recruitmentMethod:
    # recruitmentOther:
    # recruitmentNote:
    # selectionMethod:
    # selectionOther:
    # selectionNote:

    #Page 3
    # communicationMethod:
    # communicationNote:
    # participantAssumeRisk:
    # riskType:
    # riskOther:
    # riskNote:
    # willRiskChange:
    # willRiskChangeNote:

    #Page 4
    # coordinateWork:
    # coordinateWorkNote:
    # gainsharePayments:
    # gainsharePaymentsTrack:
    # gainsharePaymentsNote:
    # participantsIds:
    # participantsIdsOther:
    # participantsIDSNote:

    #Page 5
    # providerAdditionFrequency:
    # providerAdditionFrequencyOther:
    # providerAdditionFrequencyNote:
    # providerAddMethod:
    # providerAddMethodOther:
    # providerAddMethodNote:
    # providerLeaveMethod:
    # providerLeaveMethodOther:
    # providerLeaveMethodNote:
    # providerOverlap:
    # providerOverlapHierarchy:
    # providerOverlapNote:





    }) {
        id
        modelPlanID
        participants
        medicareProviderType
        statesEngagement
        participantsOther
        participantsNote
        participantsCurrentlyInModels
        participantsCurrentlyInModelsNote
        modelApplicationLevel
        expectedNumberOfParticipants
        estimateConfidence
        confidenceNote
        recruitmentMethod
        recruitmentOther
        recruitmentNote
        selectionMethod
        selectionOther
        selectionNote
        communicationMethod
        communicationNote
        participantAssumeRisk
        riskType
        riskOther
        riskNote
        willRiskChange
        willRiskChangeNote
        coordinateWork
        coordinateWorkNote
        gainsharePayments
        gainsharePaymentsTrack
        gainsharePaymentsNote
        participantsIds
        participantsIdsOther
        participantsIDSNote
        providerAdditionFrequency
        providerAdditionFrequencyOther
        providerAdditionFrequencyNote
        providerAddMethod
        providerAddMethodOther
        providerAddMethodNote
        providerLeaveMethod
        providerLeaveMethodOther
        providerLeaveMethodNote
        providerOverlap
        providerOverlapHierarchy
        providerOverlapNote
        createdBy
        createdDts
        modifiedBy
        modifiedDts
        status
    }
}
```
## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
